lib/repl.js

```diff
@@ -238,7 +238,29 @@ function REPLServer(prompt,
 
   eval_ = eval_ || defaultEval;
 
+  function preprocess(code) {
+    let cmd = code;
+    if (/^\s*\{/.test(cmd) && /\}\s*$/.test(cmd)) {
+      // It's confusing for `{ a : 1 }` to be interpreted as a block
+      // statement rather than an object literal.  So, we first try
+      // to wrap it in parentheses, so that it will be interpreted as
+      // an expression.
+      cmd = `(${cmd})`;
+      self.wrappedCmd = true;
+    } else {
+      // Mitigate https://github.com/nodejs/node/issues/548
+      cmd = cmd.replace(/^\s*function\s+([^(]+)/,
+                (_, name) => `var ${name} = function ${name}`);
+    }
+    // Append a \n so that it will be either
+    // terminated, or continued onto the next expression if it's an
+    // unexpected end of input.
+    return `${cmd}\n`;
```

Comment:
@nunocastromartins Good point. `\n` was appended for most of the cases and it was done in same `preprocess`.  I'll add `\n` for all cases as if eval function receives what user entered including the final `enter`.

CC: @Fishrock123